main.go

```diff
@@ -77,6 +58,50 @@ func createNewArticle(w http.ResponseWriter, r *http.Request) {
 	// fmt.Fprintf(w, "%+v", string(reqBody))
 }
 
+func deleteArticle(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	key := vars["id"]
+	for index, article := range Articles {
+		if article.Id == key {
+			Articles = append(Articles[:index], Articles[index+1:]...)
+		}
+	}
+}
+
+func updateArticle(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	key := vars["id"]
+	for index, article := range Articles {
+		if article.Id == key {
+			reqBody, _ := ioutil.ReadAll(r.Body)
+			var updatedArticle Article
+			json.Unmarshal(reqBody, &updatedArticle)
+			// Is there an easy way to spread prev values and override only some properties?
+			// combinedUpdatedArticle := {...Articles[index], ...updatedArticle}
+			Articles[index] = updatedArticle
+			json.NewEncoder(w).Encode(updatedArticle)
+		}
+		// how to return error if id not found?
+	}
+}
+
+type Email struct {
+	Subject string `json:"subject"`
+	Body    string `json:"body"`
+}
+
+// type Dadjoke struct {
```

Comment:
What's the difference between this Dadjoke struct and the one below? I tried both and they both worked 🤔

Reply:
those json bits are called json tags. in your case since the json is just lower case of the field name it can figure it out. But sometimes you may have something that looks like 
`type Example struct {
     Message string 'json:"confirmation_message"'
}`
in which case telling it the field helps it know what json fields to parse into your struct

Reply:
we always are explicit in our json field tags for clarity

Reply:
ah ok so the json tags should match the json and we can store it in the struct with whatever field name. thanks for clarifying!